exchanges/subscription/subscription_test.go

```diff
@@ -5,56 +5,140 @@ import (
+	t.Parallel()
+	s := &Subscription{}
+	k1, ok := s.EnsureKeyed().(*Subscription)
+	if assert.True(t, ok, "EnsureKeyed should return a *Subscription") {
+		assert.Same(t, s, k1, "Key should point to the same struct")
 	}
 	type platypus string
-	c = Subscription{
+	s = &Subscription{
 		Key:     platypus("Gerald"),
 		Channel: "orderbook",
-		Asset:   asset.Margin,
-		Pair:    currency.NewPair(currency.ETH, currency.USDC),
-	}
-	k2, ok := c.EnsureKeyed().(platypus)
-	if assert.True(t, ok, "EnsureKeyed should return a platypus") {
-		assert.Exactly(t, k2, c.Key, "EnsureKeyed should set the same key")
-		assert.EqualValues(t, "Gerald", k2, "key should have the correct value")
 	}
+	k2 := s.EnsureKeyed()
+	assert.IsType(t, platypus(""), k2, "EnsureKeyed should return a platypus")
+	assert.Equal(t, s.Key, k2, "Key should be the key provided")
 }
 
-// TestMarshalling logic test
-func TestMarshaling(t *testing.T) {
+// TestSubscriptionMarshalling ensures json Marshalling is clean and concise
+// Since there is no UnmarshalJSON, this just exercises the json field tags of Subscription, and regressions in conciseness
+func TestSubscriptionMarshaling(t *testing.T) {
 	t.Parallel()
-	j, err := json.Marshal(&Subscription{Channel: CandlesChannel})
+	j, err := json.Marshal(&Subscription{Key: 42, Channel: CandlesChannel})
 	assert.NoError(t, err, "Marshalling should not error")
 	assert.Equal(t, `{"enabled":false,"channel":"candles"}`, string(j), "Marshalling should be clean and concise")
 
 	j, err = json.Marshal(&Subscription{Enabled: true, Channel: OrderbookChannel, Interval: kline.FiveMin, Levels: 4})
 	assert.NoError(t, err, "Marshalling should not error")
 	assert.Equal(t, `{"enabled":true,"channel":"orderbook","interval":"5m","levels":4}`, string(j), "Marshalling should be clean and concise")
 
-	j, err = json.Marshal(&Subscription{Enabled: true, Channel: OrderbookChannel, Interval: kline.FiveMin, Levels: 4, Pair: currency.NewPair(currency.BTC, currency.USDT)})
+	j, err = json.Marshal(&Subscription{Enabled: true, Channel: OrderbookChannel, Interval: kline.FiveMin, Levels: 4, Pairs: currency.Pairs{currency.NewPair(currency.BTC, currency.USDT)}})
 	assert.NoError(t, err, "Marshalling should not error")
-	assert.Equal(t, `{"enabled":true,"channel":"orderbook","interval":"5m","levels":4,"pair":"BTCUSDT"}`, string(j), "Marshalling should be clean and concise")
+	assert.Equal(t, `{"enabled":true,"channel":"orderbook","pairs":"BTCUSDT","interval":"5m","levels":4}`, string(j), "Marshalling should be clean and concise")
 
 	j, err = json.Marshal(&Subscription{Enabled: true, Channel: MyTradesChannel, Authenticated: true})
 	assert.NoError(t, err, "Marshalling should not error")
 	assert.Equal(t, `{"enabled":true,"channel":"myTrades","authenticated":true}`, string(j), "Marshalling should be clean and concise")
 }
+
+// TestSubscriptionMatch exercises the Subscription MatchableKey interface implementation
+func TestSubscriptionMatch(t *testing.T) {
```

Comment:
rm

Reply:
Assuming this was related to the idea that *Subscription 's MatchableKey implementation isn't used, then see comment on *Subscription.Match